### PR TITLE
don't deadlock

### PR DESCRIPTION
**Pull request by mlsteele**

### CODE REVIEW

File: go/libkb/globals.go [hunk 1]
```diff
@@ -966,34 +966,35 @@ func (g *GlobalContext) ClearSharedDHKeyring() {
...
-			return nil
-		} else {
-			return makeNew()
-		}
-	}
+	return nil
+	// g.sharedDHKeyringMu.Lock()
+	// defer g.sharedDHKeyringMu.Unlock()
+
+	// uid := g.GetMyUID()
```
Comment (maxtaco) [inline go/libkb/globals.go]: my proposed fix is just to move `uid :=` to the top of the functino.

**Comment by oconnor663** [comment]
I certainly agree with the mission statement :-D

**Comment by mlsteele** [comment]
tough crowd, moved

File: go/libkb/globals.go [hunk 2]
```diff
@@ -966,12 +966,12 @@ func (g *GlobalContext) ClearSharedDHKeyring() {
...
 // BumpSharedDHKeyring recreates SharedDHKeyring if the uid/did changes.
 func (g *GlobalContext) BumpSharedDHKeyring() error {
-	g.sharedDHKeyringMu.Lock()
-	defer g.sharedDHKeyringMu.Unlock()
-
 	uid := g.GetMyUID()
 	did := g.Env.GetDeviceID()
 
+	g.sharedDHKeyringMu.Lock()
+	defer g.sharedDHKeyringMu.Unlock()
```
Comment (oconnor663) [inline go/libkb/globals.go]: Maybe include some comments about how important it is not to call `GetMyUID` inside of this lock?

**Comment by patrickxb** [comment]
function enter/leave trace would be helpful

**Comment by mlsteele** [comment]
added traces

**Pull request merged**